File: app/src/main/java/com/robotsandpencils/androiddaggerpracticemvp/common/CommonGreetingRepository.java
package com.robotsandpencils.androiddaggerpracticemvp.common;

import io.reactivex.Single;

/**
 * Created by pwray on 2017-11-01.
 */

public class CommonGreetingRepository {
    public Single<String> getGreeting() {
        return Single.just("Hello from CommonGreetingRepository");
    }
}

File: app/src/main/java/com/robotsandpencils/androiddaggerpracticemvp/common/CommonGreetingRepository.kt
package com.robotsandpencils.androiddaggerpracticemvp.common

import io.reactivex.Single

/**
 * Created by pwray on 2017-11-01.
 */

class CommonGreetingRepository {
    val greeting: Single<String>
        get() = Single.just("Hello from CommonGreetingRepository")
}
